hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/TestMergeIntoTable.scala

```diff
@@ -115,6 +116,72 @@ class TestMergeIntoTable extends HoodieSparkSqlTestBase with ScalaAssertionSuppo
     })
   }
 
+  test("Test MergeInto with more than once update actions") {
+    withRecordType()(withTempDir {tmp =>
+      val conf = new SparkConf().setAppName("insertDatasToHudi").setMaster("local[*]")
+      val spark = SparkSession.builder().config(conf)
+        .config("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+        .config("spark.sql.extensions", "org.apache.spark.sql.hudi.HoodieSparkSessionExtension")
```

Comment:
Is this PR still valid since master seems already got the fix?

Reply:
> Is this PR still valid since master seems already got the fix?

no，master is not fix. this pr can fix the problem for more than once update actions @danny0405

Reply:
Got it, thanks for the fix.